File: frontend-design/scripts/generate_banner_brief.py
#!/usr/bin/env python3
"""Generate a frontend banner/hero brief from local specs.

Usage:
  python scripts/generate_banner_brief.py linkedin personal_banner "Agent OS release"
  python scripts/generate_banner_brief.py website full_width_hero "Atmo Network"
"""

from __future__ import annotations

import argparse
import csv
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
SPECS = ROOT / "data" / "banner-specs.csv"


def load() -> list[dict[str, str]]:
    with SPECS.open(newline="", encoding="utf-8") as handle:
        return list(csv.DictReader(handle))


def find(surface: str, kind: str) -> dict[str, str] | None:
    s = surface.lower()
    k = kind.lower()
    for row in load():
        if s in row["surface"].lower() and (k in row["type"].lower() or k in row["surface"].lower()):
            return row
    for row in load():
        if s in row["surface"].lower():
            return row
    return None


def main() -> int:
    parser = argparse.ArgumentParser(description="Generate banner/hero brief")
    parser.add_argument("surface")
    parser.add_argument("type")
    parser.add_argument("message", nargs="+")
    parser.add_argument("--style", default="")
    parser.add_argument("--cta", default="")
    args = parser.parse_args()
    row = find(args.surface, args.type)
    message = " ".join(args.message)
    if not row:
        print("No matching banner spec found")
        return 1
    print(f"# Banner Brief: {message}\n")
    print(f"- Surface: {row['surface']} / {row['type']}")
    print(f"- Size: {row['size']} ({row['aspect_ratio']})")
    print(f"- Safe zone: {row['safe_zone']}")
    print(f"- Notes: {row['notes']}")
    if args.style:
        print(f"- Art direction: {args.style}")
    if args.cta:
        print(f"- CTA: {args.cta}")
    print("\n## Design Rules\n")
    print("- One dominant message.")
    print("- One CTA when the banner is interactive or ad-like.")
    print("- Keep essential text inside the safe zone.")
    print("- Max two type families unless the concept requires more.")
    print("- Preserve legibility at target display size and thumbnail size.")
    print("- Treat platform crop as composition, not simple scaling.")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: frontend-design/scripts/generate_banner_brief.mjs
#!/usr/bin/env node
/** Generate a frontend banner or hero brief from local specs. */

import { pathToFileURL } from "node:url";
import { loadCsv } from "./_catalog.mjs";

function parseArgs(args) {
  let style = "";
  let cta = "";
  const positional = [];
  for (let index = 0; index < args.length; index += 1) {
    if (args[index] === "--style") {
      style = args[index + 1] ?? "";
      index += 1;
    } else if (args[index] === "--cta") {
      cta = args[index + 1] ?? "";
      index += 1;
    } else positional.push(args[index]);
  }
  if (positional.length < 3) throw new Error("Usage: generate_banner_brief.mjs <surface> <type> <message...> [--style <style>] [--cta <cta>]");
  return { surface: positional[0], type: positional[1], message: positional.slice(2).join(" "), style, cta };
}

function findSpec(surface, type) {
  const rows = loadCsv("banner-specs.csv");
  const surfaceQuery = surface.toLowerCase();
  const typeQuery = type.toLowerCase();
  return rows.find((row) => row.surface.toLowerCase().includes(surfaceQuery) && (row.type.toLowerCase().includes(typeQuery) || row.surface.toLowerCase().includes(typeQuery)))
    ?? rows.find((row) => row.surface.toLowerCase().includes(surfaceQuery));
}

export function main(args = process.argv.slice(2)) {
  let options;
  try {
    options = parseArgs(args);
  } catch (error) {
    console.error(error.message);
    return 2;
  }
  const row = findSpec(options.surface, options.type);
  if (!row) {
    console.log("No matching banner spec found");
    return 1;
  }
  const lines = [
    `# Banner Brief: ${options.message}`, "",
    `- Surface: ${row.surface} / ${row.type}`,
    `- Size: ${row.size} (${row.aspect_ratio})`,
    `- Safe zone: ${row.safe_zone}`,
    `- Notes: ${row.notes}`,
  ];
  if (options.style) lines.push(`- Art direction: ${options.style}`);
  if (options.cta) lines.push(`- CTA: ${options.cta}`);
  lines.push(
    "", "## Design Rules", "",
    "- One dominant message.",
    "- One CTA when the banner is interactive or ad-like.",
    "- Keep essential text inside the safe zone.",
    "- Max two type families unless the concept requires more.",
    "- Preserve legibility at target display size and thumbnail size.",
    "- Treat platform crop as composition, not simple scaling.",
  );
  process.stdout.write(`${lines.join("\n")}\n`);
  return 0;
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) process.exitCode = main();
